src/VisualStudio/Core/Def/Implementation/PullMemberUp/SymbolDependentsBuilder.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.  
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.PullMemberUp
+{
+    internal static class SymbolDependentsBuilder
+    {
+        internal static ImmutableDictionary<ISymbol, Task<ImmutableArray<ISymbol>>> FindMemberToDependentsMap(
+            Document document,
+            ImmutableArray<ISymbol> membersInType,
+            CancellationToken cancellationToken)
+        {
+            return membersInType.ToImmutableDictionary(
+                member => member,
+                member =>
+                {
+                    var builder = new SymbolWalker(document, membersInType, member);
+                    return Task.Run(() => builder.FindMemberDependentsAsync(cancellationToken), cancellationToken);
```

Comment:
@CyrusNajmabadi 
This is the class I asked you last year :)
So put the Operation Walker as an inner class here. 
(Its purpose is to traverse the operations in a member and find the interactions between the members in the type, and it is run on different threads at line 25. Make this based on Jason's comments  and I can double check it with him when he come back)

Reply:
Thanks!

Reply:
FWIW: you could inline the 'new symbolwalker' into the lambda.  There's not much point creating it outside and capturing it.